kernel/signal.c

```diff
@@ -636,8 +636,10 @@ int_t sys_rt_sigsuspend(addr_t mask_addr, uint_t size) {
 
 int_t sys_pause() {
     lock(&current->sighand->lock);
-    while (wait_for(&current->pause, &current->sighand->lock, NULL) != _EINTR)
-        continue;
+    TASK_BLOCKING_AREA {
```

Comment:
Instead of doing this, it would make more sense to put the blocking area inside of wait_for.

Reply:
Unfortunately this might bring a deadlock. I just want this blocking defines as simple as possible, so I just put them on the most of blocking syscalls in Linux ([list of blocking sys calls](https://techpubs.jurassic.nl/manuals/0650/developer/Debugger_Ref/sgi_html/ch10.html)). From this list only fcntl and close are not wrapped with `TASK_MAY_BLOCK`, since their implementation in our case is not blocking.

Reply:
This list seems outdated? It doesn't include futex, for example.